eng/pipelines/templates/stages/archetype-sdk-client.yml

```diff
@@ -35,23 +38,25 @@ stages:
     jobs:
     - template: ../jobs/archetype-sdk-client.yml
       parameters:
-        ServiceDirectory: ${{parameters.ServiceDirectory}}
-        ToxEnvParallel: ${{parameters.ToxEnvParallel}}
-        BuildDocs: ${{parameters.BuildDocs}}
-        InjectedPackages: ${{parameters.InjectedPackages}}
-        SkipPythonVersion: ${{parameters.SkipPythonVersion}}
-        AdditionalTestMatrix: ${{parameters.AdditionalTestMatrix}}
-        DevFeedName: ${{parameters.DevFeedName}}
+        ServiceDirectory: ${{ parameters.ServiceDirectory }}
+        TestPipeline: ${{ parameters.TestPipeline }}
```

Comment:
Ah but we pass it in here. The _reason_ that this works is down to the fact that azure devops always renders parameters named the same _all the way down into the template_.

So while `TestPipeline` isn't an argument to client, it's showing up because during the initial replacement, devops doesn't _care_ where the value is coming from as long as it CAN resolve it.

Reply:
I think for clarity we probably should add it to `archetype-sdk-client.yml`

Reply:
You are right. But have added the parameter with default value to `eng/pipelines/templates/stages/archetype-sdk-client.yml` and `eng/pipelines/templates/jobs/archetype-sdk-client.yml`